ios/browser/api/history/brave_history_api.mm

```diff
@@ -0,0 +1,241 @@
+#include "ios/chrome/browser/browser_state/chrome_browser_state.h"
+#include "ios/chrome/browser/history/history_service_factory.h"
+#include "ios/chrome/browser/history/web_history_service_factory.h"
+#include "ios/web/public/thread/web_task_traits.h"
+#include "ios/web/public/thread/web_thread.h"
+#include "net/base/mac/url_conversions.h"
+#include "ui/base/page_transition_types.h"
+#include "url/gurl.h"
+
+#include "brave/ios/browser/api/history/brave_history_observer.h"
+#include "brave/ios/browser/api/history/history_service_listener_ios.h"
+
+#if !defined(__has_feature) || !__has_feature(objc_arc)
+#error "This file requires ARC support."
+#endif
+
+#pragma mark - IOSHistoryNode
+
+@interface IOSHistoryNode () {
+  std::u16string title_;
+  GURL gurl_;
+  base::Time date_added_;
+}
+@end
+
+@implementation IOSHistoryNode
+
+- (instancetype)initWithURL:(NSURL*)url
+                      title:(NSString* _Nullable)title
+                  dateAdded:(NSDate* _Nullable)dateAdded {
+  if ((self = [super init])) {
+    [self setUrl:url];
+
+    if (title) {
+      [self setTitle:title];
+    }
+
+    if (dateAdded) {
+      [self setDateAdded:dateAdded];
+    }
+  }
+
+  return self;
+}
+
+- (void)setUrl:(NSURL*)url {
+  gurl_ = net::GURLWithNSURL(url);
+}
+
+- (NSURL*)url {
+  return net::NSURLWithGURL(gurl_);
+}
+
+- (void)setTitle:(NSString*)title {
+  title_ = base::SysNSStringToUTF16(title);
+}
+
+- (NSString*)title {
+  return base::SysUTF16ToNSString(title_);
+}
+
+- (void)setDateAdded:(NSDate*)dateAdded {
+  date_added_ = base::Time::FromDoubleT([dateAdded timeIntervalSince1970]);
+}
+
+- (NSDate*)dateAdded {
+  return [NSDate dateWithTimeIntervalSince1970:date_added_.ToDoubleT()];
+}
+@end
+
+#pragma mark - BraveHistoryAPI
+
+@interface BraveHistoryAPI () {
+  // History Service for adding and querying
+  history::HistoryService* history_service_;
+  // WebhistoryService for delete operations
+  history::WebHistoryService* web_history_service_;
+  // Tracker for history requests.
+  base::CancelableTaskTracker tracker_;
+}
+@property(nonatomic, strong) void (^query_completion)(NSArray<IOSHistoryNode*>*)
+    ;
+@end
+
+@implementation BraveHistoryAPI
+
+- (instancetype)initWithBrowserState:(ChromeBrowserState*)state {
+  if ((self = [super init])) {
+    history_service_ = ios::HistoryServiceFactory::GetForBrowserState(
+        state, ServiceAccessType::EXPLICIT_ACCESS);
+    web_history_service_ =
+        ios::WebHistoryServiceFactory::GetForBrowserState(state);
+    DCHECK(history_service_);
+  }
+  return self;
+}
+
+- (void)dealloc {
+  history_service_ = nil;
+  web_history_service_ = nil;
+}
+
+- (id<HistoryServiceListener>)addObserver:(id<HistoryServiceObserver>)observer {
+  return [[HistoryServiceListenerImpl alloc] init:observer
+                                   historyService:history_service_];
+}
+
+- (void)removeObserver:(id<HistoryServiceListener>)observer {
+  [observer destroy];
+}
+
+- (bool)isLoaded {
```

Comment:
as much as possible I would try to keep the same method names (with changes for language conventions), it makes things easier to search. So `isBackendLoaded`

Reply:
Fixed